core/src/main/java/org/jsmart/zerocode/core/report/ZeroCodeReportGeneratorImpl.java

```diff
@@ -82,6 +81,26 @@ public ZeroCodeReportGeneratorImpl(ObjectMapper mapper) {
         this.mapper = mapper;
     }
 
+    /**
+     * gets unique steps from a scenario. In case of retries, the steps have same correlation id and if
+     * one of the retries is successful, we include it in the result(not the one which failed).
+     * @param steps
+     * @return
+     */
+     List<ZeroCodeReportStep> getUniqueSteps(List<ZeroCodeReportStep> steps){
+        Map<String,ZeroCodeReportStep> result = new HashMap<>();
+        steps.forEach(step->{
+            if(result.containsKey(step.getCorrelationId())){
+                if(step.getResult().equals(RESULT_PASS)){
+                    result.put(step.getCorrelationId(),step);
+                }
+            }else{
+                result.put(step.getCorrelationId(),step);
+            }
```

Comment:
I think that could be simplified with a merge, something like that :
`result.merge(step. getCorrelationId(), step, (s1, s2) -> RESULT_PASS.equals(s1.getResult()) ? s1 : s2)`

Reply:
@sparrowV , this looks clean, short and not too complicated to read!
If you could take care fo this(`result.merge(...)`) bit , please. We are almost there ✌️